DirectProgramming/DPC++FPGA/Tutorials/Features/fp_pragmas/README.md

```diff
@@ -0,0 +1,178 @@
+#Contraction and Reassociation on Floating Points
+This FPGA tutorial explains how to use the fp reassociate and fp contract pragmas for floating points.
+
+***Documentation***:  The [DPC++ FPGA Code Samples Guide](https://software.intel.com/content/www/us/en/develop/articles/explore-dpcpp-through-intel-fpga-code-samples.html) helps you to navigate the samples and build your knowledge of DPC++ for FPGA. <br>
+The [oneAPI DPC++ FPGA Optimization Guide](https://software.intel.com/content/www/us/en/develop/documentation/oneapi-fpga-optimization-guide) is the reference manual for targeting FPGAs through DPC++. <br>
+The [oneAPI Programming Guide](https://software.intel.com/en-us/oneapi-programming-guide) is a general resource for target-independent DPC++ programming.
+
+| Optimized for                     | Description
+---                                 |---
+| OS                                | Linux* Ubuntu* 18.04; Windows* 10
+| Hardware                          | Intel® Programmable Acceleration Card (PAC) with Intel Arria® 10 GX FPGA; <br> Intel® FPGA Programmable Acceleration Card (PAC) D5005 (with Intel Stratix® 10 SX)
+| Software                          | Intel® oneAPI DPC++ Compiler <br> Intel® FPGA Add-On for oneAPI Base Toolkit
+| What you will learn               | The basic usage of the `fp contract(fast|off)` and `fp reassociate(on|off)` pragmas <br> How the `fp contract(fast|off)` and `fp reassociate(on|off)` pragmas affect resource use <br> How to apply the `fp contract(fast|off)` and `fp reassociate(on|off)` pragmas in your program
+| Time to complete                  | 20 minutes
+
+
+
+## Purpose
+This tutorial demonstrates a simple example of using the `fp contract(fast|off)` and `fp reassociate(on|off)` pragmas.
+
+### Description of the pragmas
+The `fp contract(fast|off)` pragma controls whether the compiler can skip intermediate rounding and conversions mainly between double precision arithmetic operations. If multiple occurrences of this pragma affect the same scope of your code, the pragma with the narrowest scope takes precedence.
+
+The `fp reassociate(on|off)` pragma controls the relaxing of the order of floating point arithmetic operations within the code block that this pragma is applied to. If multiple occurrences of this pragma affect the same scope of your code, the pragma with the narrowest scope takes precedence.
+
+#### Example
+This tutorial design applies `fp contract(off)` and `fp reassociate(off)` at a global scope.
```

Comment:
Why is this necessary?  Is the default for these to be on?  How do these pragmas interact with command line arguments related to floating point operations?

Reply:
Yes the default for these is fp contract(fast) and fp reassociate(on) due to fast-math setting to on by default